e2e/cluster.go

```diff
@@ -57,7 +57,7 @@ func (c *Cluster) MaxPodsPerNode() (int, error) {
 	return 0, fmt.Errorf("cluster agentpool profiles were nil or empty: %+v", c.Model)
 }
 
-func prepareCluster(ctx context.Context, cluster *armcontainerservice.ManagedCluster, isAirgap, isNonAnonymousPull bool) (*Cluster, error) {
+func prepareCluster(ctx context.Context, cluster *armcontainerservice.ManagedCluster, isAirgap, isNonAnonymousPull, addFirewall bool) (*Cluster, error) {
```

Comment:
nit: the parameter count here is getting a little unwieldy, it's also difficult to tell what's being configured at the call-sites, might be worth creating a new `ClusterPreperation` struct or something which has fields for airgap nonAnonymous (I think it should be "authenticated" instead...), and addFIrewall

non-blocking, can do this later